modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -15,447 +15,423 @@
  * <p>
- * optimized by Tayeb Chlyah
+ * Note that it does not depend on a specific couchbase SDK, so it can be used with both the Java SDK 2 and 3 as well
+ * as the Scala SDK 1 or newer. We recommend using the latest and greatest SDKs for the best experience.
  */
-@AllArgsConstructor
 public class CouchbaseContainer extends GenericContainer<CouchbaseContainer> {
 
-    public static final String VERSION = "5.5.1";
-    public static final String DOCKER_IMAGE_NAME = "couchbase/server:";
-    public static final ObjectMapper MAPPER = new ObjectMapper();
-    public static final String STATIC_CONFIG = "/opt/couchbase/etc/couchbase/static_config";
-    public static final String CAPI_CONFIG = "/opt/couchbase/etc/couchdb/default.d/capi.ini";
+    private static final int MGMT_PORT = 8091;
+
+    private static final int MGMT_SSL_PORT = 18091;
+
+    private static final int VIEW_PORT = 8092;
 
-    private static final int REQUIRED_DEFAULT_PASSWORD_LENGTH = 6;
+    private static final int VIEW_SSL_PORT = 18092;
 
-    private String memoryQuota = "300";
+    private static final int QUERY_PORT = 8093;
 
-    private String indexMemoryQuota = "300";
+    private static final int QUERY_SSL_PORT = 18093;
 
-    private String clusterUsername = "Administrator";
+    private static final int SEARCH_PORT = 8094;
 
-    private String clusterPassword = "password";
+    private static final int SEARCH_SSL_PORT = 18094;
 
-    private boolean keyValue = true;
+    private static final int KV_PORT = 11210;
 
-    @Getter
-    private boolean query = true;
+    private static final int KV_SSL_PORT = 11207;
 
-    @Getter
-    private boolean index = true;
+    private static final String DOCKER_IMAGE_NAME = "couchbase/server";
 
-    @Getter
-    private boolean primaryIndex = true;
+    private static final String VERSION = "6.5.0";
 
-    @Getter
-    private boolean fts = false;
+    private static final ObjectMapper MAPPER = new ObjectMapper();
 
-    @Getter(lazy = true)
-    private final CouchbaseEnvironment couchbaseEnvironment = createCouchbaseEnvironment();
+    private static final OkHttpClient HTTP_CLIENT = new OkHttpClient()
+        .newBuilder()
+        .readTimeout(Duration.ofMinutes(1))
+        .build();
 
-    @Getter(lazy = true)
-    private final CouchbaseCluster couchbaseCluster = createCouchbaseCluster();
+    private String username = "Administrator";
 
-    private List<BucketAndUserSettings> newBuckets = new ArrayList<>();
+    private String password = "password";
 
-    private String urlBase;
+    private Set<CouchbaseService> enabledServices = EnumSet.allOf(CouchbaseService.class);
 
-    private SocatContainer proxy;
+    private final List<BucketDefinition> buckets = new ArrayList<>();
 
+    /**
+     * Creates a new couchbase container with the default image and version.
+     */
     public CouchbaseContainer() {
-        this(DOCKER_IMAGE_NAME + VERSION);
+        this(DOCKER_IMAGE_NAME + ":" + VERSION);
```

Comment:
ok, old code was awkward here :laughing:

Reply:
this is also the reason why we should not have it public - to be able to change it without having to deprecate anything :D